util/persistance.go

```diff
@@ -0,0 +1,39 @@
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+
+	"github.com/natefinch/atomic"
+	xdr "github.com/nullstyle/go-xdr/xdr3"
+)
+
+func Persist(filename string, v any) error {
+	var w bytes.Buffer
+	_, err := xdr.Marshal(&w, v)
+	if err != nil {
+		return fmt.Errorf("serializing: %w", err)
+	}
+
+	err = atomic.WriteFile(filename, &w)
+	if err != nil {
+		return fmt.Errorf("writing to disk: %w", err)
+	}
+
+	return nil
+}
+
+func Load(filename string, v any) error {
+	data, err := os.ReadFile(filename) //#nosec G304
+	if err != nil {
+		return fmt.Errorf("loading file: %w", err)
+	}
+
+	_, err = xdr.Unmarshal(bytes.NewReader(data), v)
+	if err != nil {
+		return fmt.Errorf("deserializing: %w", err)
+	}
+
+	return nil
+}
```

Comment:
Using `xdr.NewDecoder` avoids having to read the whole file into memory before decoding it. The same could be done with `xdr.NewEncoder` and then `atomic.ReplaceFile` for storing the state.

```suggestion
func Load(filename string, v any) error {
	f, err := os.Open(filename) //#nosec G304
	if err != nil {
		return fmt.Errorf("loading file: %w", err)
	}

	_, err = xdr.NewDecoder(f).Decode(v)
	if err != nil {
		return fmt.Errorf("deserializing: %w", err)
	}

	return nil
}
```

Reply:
Changed to use `os.Open()`. Kept using `xdr.Unmarshal()` as it is the exactly same thing as `xdr.NewDecoder(f).Decode(v)`.